OidcDebugger/Views/Home/Index.cshtml

```diff
@@ -1,197 +1,246 @@
+                            </p>
+                        }
+                    </info-card>
+
+                    <info-card v-show="selected === 'state'" title="State">
+                        <p>The <b>state</b> is an optional value that is carried through the whole flow and returned to the client.</p>
+                        <p>It's common to use state to store an anti-forgery token that can be verified after the login flow is complete. Another common use is storing the location the user should be redirected to after logging in.</p>
+                    </info-card>
+
+                    <info-card v-show="selected === 'nonce'" title="Nonce">
+                        <p>A <b>nonce</b> (or <b>n</b>umber used <b>once</b>) is a random value that is used to prevent <a href="https://en.wikipedia.org/wiki/Replay_attack">replay attacks</a>. </p>
+                    </info-card>
+                </div>
+            </div>
+            <div class="row">
+                <div class="col-7">
+                    <div class="md-form">
+                        <label for="responseType" class="form-group-label">
+                            Response type <span class="required-text">(required)</span>
+                        </label>
+                        <div class="form-group-inline">
+                            <div class="form-group">
+                                <input type="checkbox" id="responseType-code" value="code"
+                                       v-model="responseTypesArray" v-on:focus="showInfo" v-on:click="saveParameters" required>
+                                <label for="responseType-code" class="select-label">code</label>
+                            </div>
+                            <div class="form-group">
+                                <input type="checkbox" id="responseType-token" value="token"
+                                       v-model="responseTypesArray" v-on:focus="showInfo" v-on:click="saveParameters" required>
+                                <label for="responseType-token" class="select-label">token</label>
+                            </div>
+                            @if (Tenant.Shortname == "oidc")
+                            {
+                                <div class="form-group">
+                                    <input type="checkbox" id="responseType-id_token" value="id_token"
+                                           v-model="responseTypesArray" v-on:focus="showInfo" v-on:click="saveParameters" required>
+                                    <label for="responseType-id_token" class="select-label">id_token</label>
+                                </div>
+                            }
+                        </div>
+
+                        <div v-show="responseTypesArray.indexOf('code') > -1">
+                            <label for="pkceMethod" class="form-group-label">
+                                PKCE Method
```

Comment:
Mostly for @nbarbettini:

The term "PKCE Method" _feels_ weird, (sorry, I know that is super subjective), maybe "Use PKCE"?
(though "Use PKCE" with a value of "disabled" seems odd too).
Maybe @aaronpk chime in with thoughts?

Since the "PKCE" section is four fields, maybe they should all be in some sort of section or contained within a border?

<img width="730" alt="image" src="https://user-images.githubusercontent.com/99954/156632106-a18cca57-6732-4ef8-80ba-e2fd986dbc36.png">

Reply:
I agree @bdemers. I like the idea of having a checkbox called `Use PKCE` and a panel that expands to show the 4 PKCE-related fields if that checkbox is checked. And, if the focus is in any of the PKCE-related fields, the right-side help panel can explain what PKCE is.

Reply:
@nbarbettini @bdemers  [commit for the usePkce checkbox](https://github.com/nbarbettini/oidc-debugger/pull/72/commits/f46304e181c3211f3956db63a2b4d8c16e6bf849)

I removed the `disabled` option as its redundant when you have the "Use PKCE" checkbox and edited the JavaScript to use the new usePkce variable

Reply:
We are trying to move people away from associating PKCE with mobile apps, since it's also beneficial to web server apps and any apps that also have a client secret.